packages/core/core/src/UncommittedAsset.js

```diff
@@ -129,12 +129,10 @@ export default class UncommittedAsset {
     this.value.contentKey = contentKey;
     this.value.mapKey = mapKey;
     this.value.astKey = astKey;
-    this.value.outputHash = md5FromString(
-      [
-        this.value.hash,
-        pipelineKey,
-        await getInvalidationHash(this.getInvalidations(), this.options),
-      ].join(':'),
+    this.value.outputHash = hashString(
+      (this.value.hash ?? '') +
+        pipelineKey +
+        (await getInvalidationHash(this.getInvalidations(), this.options)),
     );
```

Comment:
...

Reply:
This one should be ok? There will always be a pipelineKey and invalidation hash.